src/safeposix/syscalls/net_calls.rs

```diff
@@ -704,27 +849,41 @@ impl Cage {
             );
         }
 
+        //In the case that the domain is AF_UNIX, AF_INET, or AF_INET6, perform a connection
+        //Otherwise, return with an error due to the domain being unsupported in lind
         match sockhandle.domain {
             AF_UNIX => self.connect_tcp_unix(&mut *sockhandle, sockfdobj, remoteaddr),
             AF_INET | AF_INET6 => self.connect_tcp_inet(&mut *sockhandle, sockfdobj, remoteaddr),
             _ => return syscall_error(Errno::EINVAL, "connect", "Unsupported domain provided"),
         }
     }
 
+    //The function sets up a connection on a TCP socket with a unix address family
+    //Args: sockhandle is a mut reference to the SocketHandle of the local socket
+    //      sockfdobj is a mut reference to the Socket Description of the local socket
+    //      remoteaddr is a reference to the remote address that the local socket will connect to 
+    //On success, zero is returned. On error, -errno is returned, and
+    //errno is set to indicate the error.
     fn connect_tcp_unix(
         &self,
         sockhandle: &mut SocketHandle,
         sockfdobj: &mut SocketDesc,
         remoteaddr: &interface::GenSockaddr,
     ) -> i32 {
         // TCP domain socket logic
+        //Check if the local address of the socket handle is not set
         if let None = sockhandle.localaddr {
+            //Assign a new local address for the Unix domain socket in sockhandle. This is necessary as
+            //each Unix domain socket needs a unique address (path) to distinguish it from other sockets.
             let localaddr = Self::assign_new_addr_unix(&sockhandle);
             self.bind_inner_socket(&mut *sockhandle, &localaddr, false);
         }
+        //Normalize the remote address to a path buffer
         let remotepathbuf = normpath(convpath(remoteaddr.path()), self);
 
-        // try to get and hold reference to the key-value pair, so other process can't alter it
+        //NET_METADATA.domsock_paths is the set of all currently bound domain sockets
+        //try to get and hold reference to the key-value pair, so other process can't alter it
+        // ** How does the line below hold a reference to the key so other processes can't alter it ? ** //
```

Comment:
I think this is the default feature offered by DashSet (underlying data structure used by domsock_paths)